Dockerfile.windows

```diff
@@ -249,19 +252,60 @@ RUN `
   Remove-Item C:\binutils.zip; `
   Remove-Item C:\gitsetup.zip; `
   `
-  Write-Host INFO: Creating source directory...; `
-  New-Item -ItemType Directory -Path ${GOPATH}\src\github.com\docker\docker | Out-Null; `
+  # Ensure all directories exist that we will require below....
+  $srcDir = """$Env:GOPATH`\src\github.com\docker\docker\bundles"""; `
+  Write-Host INFO: Ensuring existence of directory $srcDir...; `
+  New-Item -Force -ItemType Directory -Path $srcDir | Out-Null; `
   `
   Write-Host INFO: Configuring git core.autocrlf...; `
-  C:\git\cmd\git config --global core.autocrlf true; `
+  C:\git\cmd\git config --global core.autocrlf true; 
+
+  # Need a different layer  because the current layer  will not
+  # have go in its path. We could always figure that out either
+  # by observing  the current  behavior of  go-s installer (and 
+  # assuming it will not change in the future) or by  observing
+  # change in  PATH value due to the install. Instead we choose
+  # to just build a different layer - arguably an approach that
+  # is more robust.
+RUN `
+  `
+  Function Build-GoTestSum() { `
+    $GotestsumPath=Join-Path -Path $Env:GOPATH -ChildPath "src\gotest.tools\gotestsum"; `
+    Write-Host "INFO: Building gotestsum version $Env:GOTESTSUM_COMMIT in $Env:GOPATH"; `
+    $optsForGet = @('"get"', '"-d"', '"gotest.tools/gotestsum"'); ` 
+    &go $optsForGet; `
+    $savedExitCode = $LASTEXITCODE; `
+    if ($savedExitCode -ne 0) {  `
+      Throw '"Failed getting gotestsum sources..."'  `
+    }; `
+    Write-Host "INFO:     Sources obtained for gotestsum..."; `
+    Push-Location $GotestsumPath; `
+    $optsForCheckout = @('"checkout"', '"-q"', """$Env:GOTESTSUM_COMMIT"""); ` 
+    &git $optsForCheckout; `
+    $savedExitCode = $LASTEXITCODE; `
+    if ($savedExitCode -eq 0) { `
+      Write-Host "INFO:     Checkout done for gotestsum..."; `
+      $optsForBuild = @('"build"', '"-buildmode=exe"'); ` 
+      &go $optsForBuild; `
+      $savedExitCode = $LASTEXITCODE; `
+    } else { `
+      Throw '"gotestsum checkout failed..."'; `
+    } `
+    Pop-Location; `
+    `
+    if ($savedExitCode -ne 0) {  `
+      Throw '"gotestsum build failed..."'; `
+    } `
+    Write-Host "INFO:     Build done for gotestsum..."; `
+  } `
   `
-  Write-Host INFO: Completed
+  Set-Location -Path $Env:GOPATH; `
```

Comment:
The function immediately changes to `$GotestsumPath`, so likely this isn't needed.

Reply:
Removed